pkgs/applications/virtualization/virtualbox/guest-additions/default.nix

```diff
@@ -34,16 +34,14 @@ in stdenv.mkDerivation {
 
     env.NIX_CFLAGS_COMPILE = "-Wno-error=incompatible-pointer-types -Wno-error=implicit-function-declaration";
 
-    nativeBuildInputs = [ patchelf makeWrapper ];
-    buildInputs = [ virtualBoxNixGuestAdditionsBuilder ] ++ kernel.moduleBuildDependencies;
+    nativeBuildInputs = [ patchelf makeWrapper virtualBoxNixGuestAdditionsBuilder ] ++ kernel.moduleBuildDependencies;
```

Comment:
Given that we need the guest additions for the target architecture, why did you move it to `nativeBuildInputs`?

Reply:
As the guest additions are already build depending on the kernel I thought that the resulting build would be suitable for the same target architecture as well